includes/process-donation.php

```diff
@@ -1053,168 +1033,175 @@ function give_donation_form_validate_user_login() {
+                $donor_email_index = array_search(
+                    strtolower($guest_email),
+                    array_map('strtolower', $donor->emails),
+                    true
+                );
+
+                $valid_user_data['user_id'] = $donor->user_id;
+
+                // Set email to original format.
+                // @see https://github.com/impress-org/give/issues/4025
+                $valid_user_data['user_email'] = $donor->emails[$donor_email_index];
+            }
+        }
+    } else {
+        // No email.
+        give_set_error('email_empty', __('Enter an email.', 'give'));
+    }
 
-	return $valid_user_data;
+    return $valid_user_data;
 }
 
 /**
  * Register And Login New User
  *
+ * @since   1.0
+ * @unreleased sanitize user data
+ *
  * @param array $user_data User Data.
  *
  * @access  private
- * @since   1.0
- *
  * @return  integer
  */
-function give_register_and_login_new_user( $user_data = [] ) {
-	// Verify the array.
-	if ( empty( $user_data ) ) {
-		return - 1;
-	}
-
-	if ( give_get_errors() ) {
-		return - 1;
-	}
-
-	$user_args = apply_filters(
-		'give_insert_user_args',
-		[
-			'user_login'      => isset( $user_data['user_login'] ) ? $user_data['user_login'] : '',
-			'user_pass'       => isset( $user_data['user_pass'] ) ? $user_data['user_pass'] : '',
-			'user_email'      => isset( $user_data['user_email'] ) ? $user_data['user_email'] : '',
-			'first_name'      => isset( $user_data['user_first'] ) ? $user_data['user_first'] : '',
-			'last_name'       => isset( $user_data['user_last'] ) ? $user_data['user_last'] : '',
-			'user_registered' => date( 'Y-m-d H:i:s' ),
-			'role'            => give_get_option( 'donor_default_user_role', 'give_donor' ),
-		],
-		$user_data
-	);
-
-	// Insert new user.
-	$user_id = wp_insert_user( $user_args );
-
-	// Validate inserted user.
-	if ( is_wp_error( $user_id ) ) {
-		return - 1;
-	}
-
-	// Allow themes and plugins to filter the user data.
-	$user_data = apply_filters( 'give_insert_user_data', $user_data, $user_args );
-
-	/**
-	 * Fires after inserting user.
-	 *
-	 * @since 1.0
-	 *
-	 * @param int $user_id User id.
-	 * @param array $user_data Array containing user data.
-	 */
-	do_action( 'give_insert_user', $user_id, $user_data );
-
-	/**
-	 * Filter allow user to alter if user when to login or not when user is register for the first time.
-	 *
-	 * @since 1.8.13
-	 *
-	 * return bool True if login with registration and False if only want to register.
-	 */
-	if ( true === (bool) apply_filters( 'give_log_user_in_on_register', true ) ) {
-		// Login new user.
-		give_log_user_in( $user_id, $user_data['user_login'], $user_data['user_pass'] );
-	}
-
-	// Return user id.
-	return $user_id;
+function give_register_and_login_new_user($user_data = [])
+{
+    // Verify the array.
+    if (empty($user_data)) {
+        return -1;
+    }
+
+    if (give_get_errors()) {
+        return -1;
+    }
+
+    $user_args = apply_filters(
```

Comment:
@jonwaldstein This is the actual change. The rest is formatting.

Reply:
@alaca could you please try reverting the formatting changes so we can narrow the scope of what this PR is changing? I'm sure they are harmless but i'm particularly skeptical about changes in this file as heavily used

Reply:
@jonwaldstein I did, try now. Just hide the whitespace changes (my editor converted tabs to spaces 😅)